cmd/cloud-network-config-controller/main.go

```diff
@@ -234,3 +260,62 @@ func init() {
 		klog.Exit("Controller ENV variables are empty: %q: %s, %q: %s, cannot initialize controller", controllerNamespaceEnvVar, controllerNamespace, controllerNameEnvVar, controllerName)
 	}
 }
+
+func createFeatureGateAccessor(ctx context.Context, cfg *rest.Config, operatorName, deploymentNamespace, deploymentName, desiredVersion, missingVersion string, stop <-chan struct{}) (featuregates.FeatureGateAccess, error) {
+	ctx, cancelFn := context.WithCancel(ctx)
+	go func() {
+		defer cancelFn()
+		<-stop
+	}()
+
+	kubeClient, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create kube client: %w", err)
+	}
+
+	eventRecorder := events.NewKubeRecorder(kubeClient.CoreV1().Events(deploymentNamespace), operatorName, &corev1.ObjectReference{
+		APIVersion: "apps/v1",
+		Kind:       "Deployment",
+		Namespace:  deploymentNamespace,
+		Name:       deploymentName,
+	})
+
+	configClient, err := configclient.NewForConfig(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create config client: %w", err)
+	}
+	configInformers := configinformers.NewSharedInformerFactory(configClient, 10*time.Minute)
+
+	featureGateAccessor := featuregates.NewFeatureGateAccess(
```

Comment:
Looking at the docs: https://github.com/openshift/library-go/blob/be85f840097533832dea477f08ed10a574f36353/pkg/operator/configobserver/featuregates/simple_featuregate_reader.go#L30-L36
I think it would be good to add a comment stating that we will exit if any featuregate changes, something like: https://github.com/openshift/cluster-kube-apiserver-operator/blob/master/pkg/operator/starter.go#L117